### PR TITLE
Add methods to allow updating profiles prgrammatically

### PR DESCRIPTION
**Pull request by BillHiebert**
Added a few methods to the LaunchSettingsProvider which allows web projects (and others ) to add or update the existing set of profiles. This is needed so that web projects can add an IIS Express profile and provide the additional settings needed by it.

@srivatsn

### CODE REVIEW

File: src/Microsoft.VisualStudio.ProjectSystem.Managed/ProjectSystem/Debug/ILaunchSettingsProvider.cs [hunk 1]
```diff
@@ -26,6 +26,36 @@ public interface ILaunchSettingsProvider
...
+        /// </summary>
+        Task AddOrUpdateGlobalSettingAsync(string settingName, object settingContent);
+
+        /// <summary>
+        /// Removes the specified global setting and saves the settings to disk
+        /// </summary>
+        Task RemoveGlobalSettingAsync(string settingName);
+
+        /// <summary>
+        /// Sets the active profile. This just sets te propery it does not validate that the setting matches an
```
Comment (abpiskunov) [inline src/Microsoft.VisualStudio.ProjectSystem.Managed/ProjectSystem/Debug/ILaunchSettingsProvider.cs]: >te [](start = 52, length = 2)

nit: the

**Comment by abpiskunov** [comment]
>     internal class LaunchSettings : ILaunchSettings

nit: line

---
Refers to: src/Microsoft.VisualStudio.ProjectSystem.Managed/ProjectSystem/Debug/LaunchSettings.cs:10 in 975e4bc. [](commit_id = 975e4bc661cfc845f8ec3c7cbd285ac7c69c7a15, deletion_comment = False)

**Comment by abpiskunov** [comment]
>         /// <summary>

nit: line

---
Refers to: src/Microsoft.VisualStudio.ProjectSystem.Managed/ProjectSystem/Debug/LaunchSettings.cs:13 in 975e4bc. [](commit_id = 975e4bc661cfc845f8ec3c7cbd285ac7c69c7a15, deletion_comment = False)

File: src/Microsoft.VisualStudio.ProjectSystem.Managed/ProjectSystem/Debug/LaunchSettingsProvider.cs [hunk 2]
```diff
@@ -683,5 +684,130 @@ protected override void Dispose(bool disposing)
...
+                profiles = currentSettings.Profiles;
+            }
+
+            if (addToFront)
+            {
+                profiles = profiles.Insert(0, new LaunchProfile(profile));
+            }
+            else
+            {
+                // Asertion index will be set to the current count (end of list) if an existing item was not found otherwise
```
Comment (abpiskunov) [inline src/Microsoft.VisualStudio.ProjectSystem.Managed/ProjectSystem/Debug/LaunchSettingsProvider.cs]: >Asertion [](start = 19, length = 8)

Nit: asertion?

File: src/Microsoft.VisualStudio.ProjectSystem.Managed/ProjectSystem/Debug/LaunchSettingsProvider.cs [hunk 3]
```diff
@@ -683,5 +684,130 @@ protected override void Dispose(bool disposing)
...
+        /// Sets the active profile. This just sets te propery it does not validate that the setting matches an
+        /// existing profile
+        /// </summary>
+        public async Task SetActiveProfileAsync(string profileName)
+        {
+            var props = await CommonProjectServices.ActiveConfiguredProjectProperties.GetProjectDebuggerPropertiesAsync().ConfigureAwait(false);
+            await props.ActiveDebugProfile.SetValueAsync(profileName).ConfigureAwait(true);
+        }
+
     }
```
Comment (abpiskunov) [inline src/Microsoft.VisualStudio.ProjectSystem.Managed/ProjectSystem/Debug/LaunchSettingsProvider.cs]: nit: line

**Comment by abpiskunov** [comment]
:shipit:

**Comment by BillHiebert** [comment]
Pushed a new commit to address Anton's feedback

**Pull request merged**